### ISSUE TITLE
`app.json_encoder` usage is deprecated and will be removed in Flask 2.3.0

### ISSUE THREAD

**Issue by djhoese** [skozlovf/flask-json#10]
See https://github.com/pallets/flask/pull/4692

Running my tests I see these warnings:

```
../../../miniconda3/envs/rain_py311/lib/python3.11/site-packages/flask_json.py:547
  /home/davidh/miniconda3/envs/rain_py311/lib/python3.11/site-packages/flask_json.py:547: DeprecationWarning: 'app.json_encoder' is deprecated and will be removed in Flask 2.3. Customize 'app.json_provider_class' or 'app.json' instead.
    app.json_encoder = self._encoder_class

../../../miniconda3/envs/rain_py311/lib/python3.11/site-packages/flask_json.py:674
  /home/davidh/miniconda3/envs/rain_py311/lib/python3.11/site-packages/flask_json.py:674: DeprecationWarning: 'app.json_encoder' is deprecated and will be removed in Flask 2.3. Customize 'app.json_provider_class' or 'app.json' instead.
    self._app.json_encoder = JSONEncoderWithHook
```

**Comment by skozlovf** [comment]
@djhoese Yep, I plan to rework internals to switch to the new flask API.

**Comment by skozlovf** [comment]
Fixed in `0.4.0`.
